specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2024-04-01-preview/BusinessApplicationAgents.json

```diff
@@ -0,0 +1,1712 @@
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Systems_Delete": {
+            "$ref": "./examples/businessApplicationAgents/systems/Systems_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/businessApplicationAgents/{agentResourceName}/systems": {
+      "get": {
+        "tags": [
+          "systems"
+        ],
+        "description": "ListAll the systems.",
+        "operationId": "Systems_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "in": "path",
+            "name": "agentResourceName",
+            "description": "Business Application Agent Name",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z0-9,-]*$",
+            "minLength": 3,
+            "maxLength": 63
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataFilter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataSkipToken"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok",
+            "schema": {
+              "$ref": "#/definitions/SystemsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Systems_GetAll": {
+            "$ref": "./examples/businessApplicationAgents/systems/Systems_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/businessApplicationAgents/{agentResourceName}/systems/{systemResourceName}/listActions": {
```

Comment:
>listActions [](http://example.com/codeflow?start=251&length=11)

Looks like you have modelled these as a bunch of POST actions : listactions, undoactions and reportactionstatus

Any reason why you have not modelled these with PUT\GET\DELETE semantics ?

Also how are these actions trigerred ? is that not a client operation ?

Reply:
hi @raosuhas 
thanks for the comment 

indeed - the creation of the action happens internally (as part of automatic mechanism that we have) 
the user is not creating it. 
the user can only list the actions, update their status, and undo (which is similar to update but for another field)

the updates are only support updating specific fields and not the whole instance.